added prims algorithm issue-3304

kavyasree-123 opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.

Help us understand your motivation by explaining why you decided to make this change.
You can learn more about contributing to NeoAlgo here: https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md

Happy Contributing!

-->

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
yes i have

### Description
prims algorithm is being added to cpp ds files
Prim's Algorithm is used to find the minimum spanning tree from a graph. Prim's algorithm finds the subset of edges that includes every vertex of the graph such that the sum of the weights of the edges can be minimized.
time_complexity_O(ElogV) where v no of vertices and e no of edges

### Checklist

- [yes ] I've read the contribution guidelines.
- [yes ] I've checked the issue list before deciding what to submit.
- [ yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

[#3304](https://github.com/TesseractCoding/NeoAlgo/issues/3304)  added code to this issue please kindly verify

kavyasree-123 committed:
added prims algorithm issue-3304

kavyasree-123 commented:
please review my pr and merge prim's algorithm
@bhav09 , @siddharth25pandey @HarshCasper , @ankitaggarwal23

ankitaggarwal23 reviewed (changes requested):
Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top
- Use proper code Format 
- Remove extra lines of code and spaces
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code

kavyasree-123 committed:
updated prims algorithm

added clear description
removed spaces in between
all prescribed chages are done

kavyasree-123 commented:
I have added multi line commands added description and all the others are done. please review

kavyasree-123 committed:
updated

siddharth25pandey commented on C-Plus-Plus/ds/prims algo.cpp:
```diff
@@ -0,0 +1,79 @@
+has the minimum sum of weights among all the trees that can be formed from the graph.
+(issue-3304)*/
+#include <iostream>
+#include <cstring>
```
Write single header file **bits/stdc**

kavyasree-123 replied:
yes i have changed the header file

siddharth25pandey commented on C-Plus-Plus/ds/prims algo.cpp:
```diff
@@ -0,0 +1,79 @@
+  {76,89,94,15,14,28,27,78,74},
+  {1,75,48,75,48,63,15,4,8},
+  {3,37,36,18,25,2,9,29,43}
+};
```
Take dynamic input from the user.

kavyasree-123 replied:
yes i have changed the code took input from user

siddharth25pandey commented on C-Plus-Plus/ds/prims algo.cpp:
```diff
@@ -0,0 +1,79 @@
+  memset(array, false, sizeof(array));
+  all_edges = 0;
+  array[0] = true;
+  int row , coloumn; 
```
Display message, what are you taking input from the user

kavyasree-123 replied:
done i have described

siddharth25pandey reviewed (changes requested):
@kavyasree-123 Do the suggested changes and update the **Readme** file changes

kavyasree-123 committed:
updated the declared changes

kavyasree-123 commented:
@siddharth25pandey i have made the changes and updated the code please verify

siddharth25pandey commented:
@kavyasree-123 Please check carefully, what I wrote earlier. Have a look [here](https://github.com/TesseractCoding/NeoAlgo/pull/3542#pullrequestreview-616049464)+
**Please update the Readme file details**

kavyasree-123 committed:
updated link in  readme

kavyasree-123 committed:
updated readme

kavyasree-123 committed:
Rename prims algo.cpp to prims_algo.cpp

kavyasree-123 committed:
Merge branch 'master' into issue-3304

kavyasree-123 commented:
hello i have updated readme file please verify @siddharth25pandey

anushkrishnav commented:
@siddharth25pandey Looks good to me

siddharth25pandey reviewed (approved):
@kavyasree-123 Able to merge, Thanks!

siddharth25pandey commented:
@all-contributors please add @kavyasree-123 for her code

kavyasree-123 commented:
thank you so much for all mentors..
